torch/csrc/jit/script/lexer.h

```diff
@@ -185,6 +186,36 @@ struct SharedParserData {
     *len = endptr - startptr;
     return *len > 0;
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len)
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len;
+  }
+
+  // python conconcatenates all adjacent strings "a" "b" == "ab"
+  // strings can be enclosed with 1 or 3 single or double quotes
+  // if enclosed 3 quotes newlines are valid
+  // as elsewhere, backslash and new line should be ignored
+  bool isString(const std::string& str, size_t start, size_t* len) {
+    char quote = str[start];
+    if (quote != '\"' && quote != '\'')
+      return false;
+    int quote_len = isCharCount(quote, str, start, 3) ? 3 : 1;
+    size_t end = start + quote_len;
+    while(end < str.size() && !isCharCount(quote, str, end, quote_len)) {
+      if (str[end] == '\n' && quote_len != 3) {
+        return false;
```

Comment:
Not sure what's the conventional way of handling this in compilers (@zdevito?) but this really looks like an error to me. You've successfully entered a string literal token, and then got an unexpected newline! It's not like quotes can represent anything else, so this is a clear error! Raising it in here will let us give an error message nicer than the generic "can't lex this thing".

Reply:
This is probably fine. The reason we can't report an error here is because the error report infrastructure exists above this level of abstraction. The error will just get reported as unknown token at the higher level.